File: apps/agents/src/thread-title/state.ts
import { Annotation, MessagesAnnotation } from "@langchain/langgraph";
import { ArtifactV3 } from "@opencanvas/shared/types";

export const TitleGenerationAnnotation = Annotation.Root({
  /**
   * The chat history to generate a title for
   */
  ...MessagesAnnotation.spec,
  /**
   * The artifact that was generated/updated (if any)
   */
  artifact: Annotation<ArtifactV3 | undefined>,
});

export type TitleGenerationReturnType = Partial<
  typeof TitleGenerationAnnotation.State
>;

File: agents-py/agents/src/thread_title/state.py
from typing import List, Optional, Dict, Any
from pydantic import BaseModel, Field
from langchain_core.messages import BaseMessage
from shared.src.types import ArtifactV3

class TitleGenerationState(BaseModel):
    """State representation for Title Generation graph"""
    
    messages: List[BaseMessage] = Field(
        ...,
        description="The chat history to generate a title for"
    )
    artifact: Optional[ArtifactV3] = Field(
        default=None,
        description="The artifact that was generated/updated (if any)"
    )

TitleGenerationReturnType = Dict[str, Any] 